### PR TITLE
OIDC RP-Initiated Logout POST method support

### LINKED ISSUE #11958: Keycloak does not support POST method mentioned in latest OIDC RP-Initiated Logout specs draft
### Describe the bug

OpenID Connect RP-Initiated Logout specifies that "OpenID Providers MUST support the use of the HTTP **GET and POST** methods defined in [RFC 7231](https://openid.net/specs/openid-connect-rpinitiated-1_0.html#RFC7231) [RFC7231] at the Logout Endpoint." : https://openid.net/specs/openid-connect-rpinitiated-1_0.html#RPLogout

But Keycloak only supports GET method for the official OIDC Logout enpoint specification. 

The POST method of the logout endpoint requires client authentication and a refresh token which is not compliant with the OIDC specification.

### Version

Keycloak 18.0.0

### Expected behavior

The Logout Endpoint complies with OIDC RP-Initiated Logout specification when using POST method

### Actual behavior

The Logout Endpoint doe not comply with OIDC RP-Initiated Logout specification when using POST method

### How to Reproduce?

1) Authenticate on Keycloak to get a valid ID Token
2) POST the following request : 
curl -X POST --location "http://127.0.0.1:8080/realms/master/protocol/openid-connect/logout" \
    -d "id_token_hint=<your_id_token>"

You get a response : 

HTTP/1.1 400 Bad Request
Referrer-Policy: no-referrer
X-Frame-Options: SAMEORIGIN
Strict-Transport-Security: max-age=31536000; includeSubDomains
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Content-Type: application/json
content-length: 75

{
  "error": "invalid_client",
  "error_description": "Invalid client credentials"
}

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by mposolda**
Closes #11958

### CODE REVIEW

**Comment by mposolda** [comment]
@pedroigor @mabartos Do you please have a chance to review this small PR? Thanks

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/RPInitiatedLogoutTest.java [hunk 1]
```diff
@@ -721,6 +725,51 @@ public void logoutWithBadClientId() {
...
+            String sessionId = tokenResponse.getSessionState();
+
+            Map<String, String> postParams = new HashMap<>();
+            postParams.put(OIDCLoginProtocol.POST_LOGOUT_REDIRECT_URI_PARAM, redirectUri);
+            postParams.put(OIDCLoginProtocol.ID_TOKEN_HINT, idTokenString);
+            postParams.put(OAuth2Constants.STATE, "my-state");
+            URLUtils.sendPOSTRequestWithWebDriver(oauth.getLogoutUrl().build(), postParams);
+
+            events.expectLogout(tokenResponse.getSessionState()).detail(Details.REDIRECT_URI, redirectUri).assertEvent();
+            Assert.assertThat(false, is(isSessionActive(sessionId)));
```
Comment (mabartos) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/RPInitiatedLogoutTest.java:747]: Please, always use `org.hamcrest.MatcherAssert.assertThat` instead. But, it's not a blocker for this PR, so you can keep it; it's just a nitpick.

**Review by mabartos — APPROVED** [review]
@mposolda LGTM. 

Just one nitpick for the future regarding the testsuite :)

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/RPInitiatedLogoutTest.java (hunk 1 above)
Comment (mposolda) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/RPInitiatedLogoutTest.java:747]: @mabartos Good point. I will fix that in the follow-up PR (as there is another PR related to logout I am going to send).

**Pull request merged**